### PR TITLE
Update wiki for CREATURE_FLAG_EXTRA_DONT_OVERRIDE_SAI_ENTRY

### PR DESCRIPTION
**Pull request by Gultask**
https://github.com/azerothcore/azerothcore-wotlk/pull/15976

### Description

- This also provides a new standard, which is using rows 1000+ for GUID-specific that does not override Entry SAI

Why 1000+?
Because no script on their right mind reaches that value, and it was convenient.
I thought about *100 to simulate Actionlist numbering but in reality it's not feasible, as I've had scripts that reached the high 30s, so 100+ row scripts are not far off

### Related Issue

Closes

## Thank you for contributing to the AzerothCore wiki.

Remember that the wiki is currently available in English and Spanish.

- https://www.azerothcore.org/wiki/home
- https://www.azerothcore.org/wiki/es/home

### CODE REVIEW

File: docs/smart_scripts.md [hunk 1]
```diff
@@ -59,6 +59,12 @@ The `smart_scripts` table has 30 attributes. It serves to make scripts in SQL la
 - EntryOrGuid < 0: `guid` of the creature / game object / etc.
 - **Depends on source\_type.**
 
+CREATURE_FLAG_EXTRA_DONT_OVERRIDE_SAI_ENTRY in creature_template allows us to not require duplicating rows that are shared between all creatures of the same entry in the GUID script.
+
+For example, you can keep all movement-related scripting in the GUID script, while combat scripting is handled by the ENTRY script.
+
+For creatures that use this flag, SAI row IDs cannot overlap, thus we should use row IDs starting with 1000 and incrementing it for each row (e.g. 1000, 1001, 1002, ...).
```
Comment (Kitzunu) [inline docs/smart_scripts.md:66]: This probably fits better somewhere else. Or at best can be less wordy

Comment (Gultask) [reply docs/smart_scripts.md:66]: I thought the same but couldn't figure out where if not another section.

As for it being wordy I agree but I need to explain as this adds a new rule (1000+ IDs) and can significantly change SAI usage

**Comment by Kitzunu** [comment]
resolve conflict

**Comment by Gultask** [comment]
The conflict editor is messy

**Pull request merged**